packages/zeego/src/menu/types.ts

```diff
@@ -56,39 +57,38 @@ export type MenuItemProps = (
   key: string
 }
 
-export type MenuItemIconProps = {
+export interface MenuItemCommonProps {
   /**
    * The name of an iOS-only SF Symbol. For a full list, see https://developer.apple.com/sf-symbols/.
    *
    * @platform ios
    */
   iosIconName?: string
+  /**
+   * Custom configuration for the SF Symbol icon provided by `iosIconName`. This can be used for iOS15+
+   * features like weight, scale, colors etc. The options correspond to the ImageSystemConfig in
+   * react-native-ios-context-menu
+   *
+   * @platform ios
+   */
+  iosIconConfiguration?: ImageSystemConfig
```

Comment:
Let's do an explicit `Pick` of the fields we're using:

```ts
Pick<ImageSystemConfig, 'field1', 'field2'> & {
  name: string
}
```

Reply:
And for v1, I'll want to close #9 by adding autocomplete for the `name` field there.

Reply:
Hey, instead of using `Pick` i opted for `ImageSystemSymbolConfiguration` exported in the latest version of `react-native-ios-context-menu (which means I had to bump its version), which allows the user to use all customization options. I don't think there's a need to restrict them to only a few options.